test/test_torch.py

```diff
@@ -6100,6 +6102,12 @@ def test_pow(self, device):
                         res2[i] = math.pow(m1[4][i], num)
                     self.assertEqual(res1, res2)
 
+                    # scalar ** tensor to enforce correct handling of dtypes for __rpow__().
```

Comment:
Do we ever check scalar ** tensor with negative values in the tensor, especially negative integer values?
It looks like this PR also changes how tensor ** tensor integer power works. Is there test coverage here for that?

Reply:
m1 is a multi-dimensional tensor, so m1[4] is still a random tensor for each dtype (which presumably contains negative values). existing/older tests below have `scalar ** tensor`.